test/dns_dnscmd/dnscmd_test.rb

```diff
@@ -13,63 +31,98 @@ def test_dnscmd_provider_initialization
-    assert Proxy::Dns::Dnscmd::Record.new.create_ptr_record('host.domain', '192.168.33.33')
+    assert @server.create_ptr_record('host.foo.bar.domain.local', '192.168.33.33')
   end
 
   def test_overwrite_ptr_record
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('192.168.33.33').returns('host.domain')
-    Proxy::Dns::Dnscmd::Record.new.create_ptr_record('host.domain', '192.168.33.33')
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('192.168.33.33').returns('host.foo.bar.domain.local')
+    @server.create_ptr_record('host.foo.bar.domain.local', '192.168.33.33')
   end
 
   def test_create_duplicate_ptr_record_fails
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('192.168.33.33').returns('another.host.domain')
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('192.168.33.33').returns('another.host.foo.bar.domain.local')
     assert_raise Proxy::Dns::Collision do
-      Proxy::Dns::Dnscmd::Record.new.create_ptr_record('host.domain', '192.168.33.33')
+      @server.create_ptr_record('host.foo.bar.domain.local', '192.168.33.33')
     end
   end
 
   def test_remove_address_record
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('host.domain').returns(true)
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:execute).with('/RecordDelete domain host.domain. A /f', anything).returns(true)
-    assert Proxy::Dns::Dnscmd::Record.new.remove_a_record('host.domain')
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('host.foo.bar.domain.local').returns(true)
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:execute).with('/RecordDelete bar.domain.local host.foo.bar.domain.local. A /f', anything).returns(true)
+    assert @server.remove_a_record('host.foo.bar.domain.local')
   end
 
   def test_remove_non_existent_address_record_raises_exception
-    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('host.domain').returns(false)
+    Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('host.foo.bar.domain.local').returns(false)
     assert_raise Proxy::Dns::NotFound do
-      Proxy::Dns::Dnscmd::Record.new.remove_a_record('host.domain')
+      @server.remove_a_record('host.foo.bar.domain.local')
     end
   end
 
   def test_remove_ptr_record
     Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('192.168.33.33').returns(true)
-    assert Proxy::Dns::Dnscmd::Record.new.remove_ptr_record('192.168.33.33')
+    assert @server.remove_ptr_record('192.168.33.33')
   end
 
   def test_remove_nonexistent_ptr_record_raises_exception
     Proxy::Dns::Dnscmd::Record.any_instance.expects(:dns_find).with('192.168.33.33').returns(false)
     assert_raise Proxy::Dns::NotFound do
-      Proxy::Dns::Dnscmd::Record.new.remove_ptr_record('192.168.33.33')
+      @server.remove_ptr_record('192.168.33.33')
     end
   end
+
+  def test_dns_zone_matching_edge_cases
```

Comment:
Can you rename the test so it's clear that it verifies that the longest available zone has been matched?